the-collab-lab/tcl-22-smart-shopping-list
# routing

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

Adding React routing feature for user to navigate between List view and Add item view so that the interface is clearly delinteated between viewing and adding.

## Related Issue

Closes [issue #3](https://github.com/the-collab-lab/tcl-22-smart-shopping-list/issues/3)

## Acceptance Criteria


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before




### After

![screencapture-tcl-22-shopping-list-pr18-dt-kk-routing-hbqhe1x3-web-app-list-2021-04-05-16_59_24](https://user-images.githubusercontent.com/68360696/113626764-90fc9100-9630-11eb-92bd-a5b484d9bd46.png)


## Testing Steps / QA Criteria

To test the feature, use one of the following ways:

1) You can either follow the link to see the preview: https://tcl-22-shopping-list--pr18-dt-kk-routing-hbqhe1x3.web.app
Or launch the app locally from your terminal:
 • Pull down this branch with `git pull origin dt-kk-routing` and check that branch out with `git checkout origin dt-kk-routing`
 • Run `npm install`, then `npm start`  to launch the app in your browser at [http://localhost:3000](http://localhost:3000)

2) At the bottom of the screen, you will see the links to the two views: one for the “list” view, the other for the “add an item” view. As you switch between views, the corresponding link becomes bold, the slug updates to match the view and different h2 elements are displayed. You can also switch between views by manually adding "/list" or "/additem" to the URL.
